engine/gamesys/src/gamesys/gamesys_private.cpp

```diff
@@ -83,8 +83,7 @@ namespace dmGameSystem
                 value = &value[value_index];
             }
 
-            out_desc.m_ArraySize = constant_array_size;
-            out_desc.m_IsArray   = constant_array_size > 1;
+            out_desc.m_ValueType   = (constant_array_size > 1) ? dmGameObject::PROP_VALUE_ARRAY : 0;
```

Comment:
We only have "variable"/"array" support in shaders.
So, shouldn't this always be PROP_VALUE_ARRAY ?